avocado/utils/vfio.py

```
@@ -0,0 +1,190 @@
+    :type vfio_api_version: int
+    :param vfio_check_extension:  ioctl to check iommu extension support
+    :type vfio_check_extension: int
+    :param vfio_type_iommu: expected vfio iommu type
+    :type vfio_type_iommu: int
+    :raises ValueError: if the vfio api version is invalid or if the required
+                        iommu extension is not supported.
+    :return: True if vfio container fd check passes.
+    :rtype: bool
+    """
+    if ioctl(container_fd, vfio_get_api_version) != vfio_api_version:
+        raise ValueError("Failed to get right API version")
```

Ruff (0.14.0)

Avoid specifying long messages outside the exception class

(TRY003)